tests/plots/test_declarative.py

```diff
@@ -46,6 +46,49 @@ def test_declarative_image():
     return pc.figure
 
 
+@needs_cartopy
+def test_declarative_three_dims_error():
+    """Test making an image plot with three dimensions."""
+    data = xr.open_dataset(get_test_data('narr_example.nc', as_file_obj=False))
+
+    img = ImagePlot()
+    img.data = data
+    img.colormap = 'coolwarm'
+
+    panel = MapPanel()
+    panel.plots = [img]
+
+    pc = PanelContainer()
+    pc.panel = panel
+
+    with pytest.raises(ValueError):
```

Comment:
Since we're already tweaking, how about we be specific (just in case something else fails):
```suggestion
    with pytest.raises(ValueError, match='subset for plotting'):
```